packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/InvoiceCancelModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { useRootElements } from '@/core/context/RootElementsProvider';
+import {
+  useCancelReceivableById,
+  useReceivableById,
+} from '@/core/queries/useReceivables';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Skeleton,
+} from '@mui/material';
+
+interface InvoiceCancelModalProps {
+  invoiceId: string;
+  open: boolean;
+  onClose: () => void;
+}
+
+export const InvoiceCancelModal = ({
+  invoiceId,
+  open,
+  onClose,
+}: InvoiceCancelModalProps) => {
+  const { i18n } = useLingui();
+  const { root } = useRootElements();
+  const { data: receivable, isLoading: isReceivableLoading } =
+    useReceivableById(invoiceId);
+  const cancelReceivableMutation = useCancelReceivableById(invoiceId);
+
+  const creditNoteIds =
+    receivable?.type === 'invoice'
+      ? receivable.related_documents.credit_note_ids
+      : undefined;
+
+  const hasNoCreditNotes = !creditNoteIds?.length;
```

Comment:
We must check, if the Credit Note has `draft` status. Only in this case (for the modal) we can't cancel the Invoice.